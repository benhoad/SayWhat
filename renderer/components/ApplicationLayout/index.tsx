import useApplication from "../../hooks/useApplication";
import SequencesList from "../SequencesList";
import Sequence from "../Sequence";
import Drawer from "../Drawer";

export default function Layout() {
  const { userInterface, closeSequencesList } = useApplication();

  return (
    <>
      <Drawer
        side="Left"
        isOpen={userInterface.isSequencesListOpen}
        onClose={() => closeSequencesList()}
        data-testid="drawer">
        <SequencesList />
      </Drawer>

      <Sequence />
    </>
  );
}
