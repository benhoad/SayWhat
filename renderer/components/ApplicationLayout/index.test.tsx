import { render, fireEvent } from "@testing-library/react";

import ApplicationLayout from ".";

const mockCloseSequencesList = jest.fn();
jest.mock("../../hooks/useApplication", () => {
  return () => ({
    userInterface: {
      isSequencesListOpen: true
    },
    closeSequencesList: mockCloseSequencesList
  });
});

jest.mock("../Drawer", () => {
  return (props: any) => <div data-testid={props["data-testid"]} onClick={props.onClose} />;
});

describe("ApplicationLayout", () => {
  it("renders", () => {
    expect.hasAssertions();

    const { container, queryByTestId } = render(<ApplicationLayout />);
    expect(container.firstElementChild).toMatchSnapshot();

    expect(mockCloseSequencesList).not.toHaveBeenCalled();
    fireEvent.click(queryByTestId("drawer"));
    expect(mockCloseSequencesList).toHaveBeenCalled();
  });
});
