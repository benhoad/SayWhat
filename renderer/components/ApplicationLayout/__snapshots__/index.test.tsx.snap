// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationLayout renders 1`] = `
<div
  data-testid="drawer"
/>
`;
